### PR TITLE
teleterm_test.go: Make sure daemon has started before stopping it

### LINKED ISSUE #13999: `teleterm.TestStart` flakiness
#### Link(s) to logs

- https://console.cloud.google.com/cloud-build/builds;region=us-west1/24e2eeba-faec-4115-ac7f-e4f81b0fa36f?project=ci-account

#### Relevant snippet

```
FAIL: github.com/gravitational/teleport/lib/teleterm.TestStart
===================================================
OUTPUT github.com/gravitational/teleport/lib/teleterm.TestStart
===================================================
=== RUN   TestStart
time="2022-06-29T22:20:23Z" level=info msg="tsh daemon is listening on unix:///tmp/TestStart3333087946/001/teleterm.sock."
time="2022-06-29T22:20:23Z" level=info msg="Context closed, stopping service."
    teleterm_test.go:45: 
        	Error Trace:	teleterm_test.go:45
        	            				teleterm_test.go:48
        	Error:      	Received unexpected error:
        	            	shutting down due to API Server error
        	            		grpc: the server has been stopped
        	Test:       	TestStart
--- FAIL: TestStart (0.01s)
```

### PR DESCRIPTION
**Pull request by ravicious**
Fixes #13999.

That instance of a flaky tests has thrown "grpc: the server has been stopped". It seems [this error](https://github.com/grpc/grpc-go/blob/v1.46.0/server.go#L708-L710) is thrown by google.golang.org/grpc [only when `Server.Serve` is called after `Server.Stop`](
https://github.com/grpc/grpc-go/blob/v1.46.0/server.go#L737-L746).

`teleterm.Start` is written in a way that makes it possible for `apiServer.Stop` to be called after `apiServer.Serve` and `apiServer` is just a thin wrapper over that gRPC server.

I don't know if it's possible to orchestrate this code in a way which guarantees that `Stop` is called after `Serve`. `Serve` blocks until the server is stopped. But from what I can see, even in grpc-go tests [they resorted to calling `time.Sleep`](https://github.com/grpc/grpc-go/blob/v1.46.0/server_test.go#L66-L84) so that's what I'm going to do in our tests as well.

### CODE REVIEW

**Review by marcoandredinis — APPROVED** [review]

**Comment by zmb3** [comment]
This doesn't actually "make sure that daemon has started", right? It just waits longer to increase the chances that it has started.

Is there anything more reliable than a sleep? Some sort of notification from the daemon that it has started up?

**Comment by ravicious** [comment]
@zmb3 You're right. See [my original question on the development channel](https://gravitational.slack.com/archives/C08HF8E9F/p1656688178919159). [The gRPC server](https://pkg.go.dev/google.golang.org/grpc#Server) doesn't seem to implement any kind of API for that. I don't really know how people usually handle tests for APIs like that in Go so I  followed what gRPC maintainers did.

I suppose maybe I could start the server, send some kind of an echo/heartbeat request (if possible), wait for the response and only then attempt to close it?

**Review by zmb3 — APPROVED** [review]
Thanks for clarifying. If a sleep is good enough for grpc-go I'm fine with it here.

**Pull request merged**
